### PR TITLE
chore: update jest to v27

### PR DESCRIPTION
**Pull request by snitin315**
**What kind of change does this PR introduce?**
chore

**Did you add tests for your changes?**
NA
**If relevant, did you update the documentation?**
NA
**Summary**

Update jest to v27.

**Does this PR introduce a breaking change?**
No

**Other information**
No

### CODE REVIEW

File: jest.config.js [hunk 1]
```diff
@@ -9,6 +9,7 @@ const ignorePattern =
 module.exports = {
     testPathIgnorePatterns: ignorePattern,
     testEnvironment: "node",
+    testRunner: "jest-jasmine2",
```
Comment (snitin315) [inline jest.config.js:12]: As per https://jestjs.io/blog/2021/05/25/jest-27#flipping-defaults

Comment (anshumanv) [reply jest.config.js:12]: why? We should use `jest-circus`

Comment (snitin315) [reply jest.config.js:12]: Actually we rely on jasmine, maybe we can update I will look into it.

https://github.com/webpack/webpack-cli/blob/7dad5817ca3a39a5782411d3629512ac7dc161a5/setupTest.js#L5

Comment (anshumanv) [reply jest.config.js:12]: oh yep, we need to migrate as jasmine will soon be deprecated

Comment (alexander-akait) [reply jest.config.js:12]: @snitin315 Let's merge it as is and do `jest-circus` migration in the separate PR

Comment (snitin315) [reply jest.config.js:12]: Will do 👍

**Review by anshumanv — APPROVED** [review]

**Pull request merged**